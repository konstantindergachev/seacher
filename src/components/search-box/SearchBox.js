import React from 'react';
import './SearchBox.scss';

const SearchBox = ({ placeholder, handleChange }) => {
  return (
    <input
      type="search"
      className="search"
      placeholder={placeholder}
      onChange={handleChange}
    />
  );
};

export default SearchBox;
