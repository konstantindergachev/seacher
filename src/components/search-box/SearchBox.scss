.search {
  border: 1px solid grey;
  border-radius: 5px;
  outline: transparent;
  line-height: 30px;
  // text-transform: uppercase;
  padding: 5px 10px;
  margin: 30px 0;
  width: 50%;
}
