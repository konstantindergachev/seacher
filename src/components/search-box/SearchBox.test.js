import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import SearchBox from './SearchBox';

let wrapper;
beforeEach(() => {
  const mockProps = {
    placeholder: 'test name',
    handleChange: jest.fn(),
  };
  wrapper = shallow(
    <SearchBox
      placeholder={mockProps.placeholder}
      handleChange={mockProps.handleChange}
    />
  );
});

describe('SearchBox', () => {
  it('to render SearchBox component', () => {
    expect(shallow(<SearchBox />).length).toEqual(1);
  });

  it('to get the snapshot of SearchBox component', () => {
    const tree = renderer.create(<SearchBox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
