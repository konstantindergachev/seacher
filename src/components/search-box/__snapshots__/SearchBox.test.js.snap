// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox to get the snapshot of SearchBox component 1`] = `
<input
  className="search"
  type="search"
/>
`;
